tests/queries/0_stateless/03611_null_safe_comparsion.sql

```diff
@@ -0,0 +1,360 @@
+    `c_datetime` DateTime,
+    `c_dt64` DateTime64(3),
+    `c_string` String,
+    `c_fstring` FixedString(4),
+    `c_enum8` Enum8('a' = 1, 'b' = 2, '' = 0),
+    `c_enum16` Enum16('x' = 100, 'y' = 200, '' = 0),
+    `c_array` Array(Int32),
+    `c_tuple` Tuple(Int32,String),
+    `c_map` Map(String, Int32),
+    `c_nullable` Nullable(Int32),
+    `c_uuid` UUID,
+    `c_ipv4` IPv4,
+    `c_ipv6` IPv6,
+    `c_json` JSON
+)
+ENGINE = MergeTree
+ORDER BY c_int32;
+
+SELECT 'Insert rows containing:';
+SELECT '• normal values';
+SELECT '• NULLs';
+SELECT '• NaN';
+SELECT '• edge numeric boundaries';
+INSERT INTO 03611_nscmp_tbl VALUES
+(
+    -- Normal row with all valid values
+    1,1,1,1,
+    1,1,1,1,
+    1.0,1.0,
+    123.4567,
+    '2025-09-22',
+    '2025-09-22 12:34:56',
+    '2025-09-22 12:34:56.789',
+    'abc','abcd','a','x',
+    [1,2,3],
+    (1,'t'),
+    map('k1',1),
+    100,
+    generateUUIDv4(),
+    '127.0.0.1',
+    '::1',
+    '{"k":"v"}'
+),
+(
+    -- Row with all NULLs to test NULL-safe comparisons
+    NULL,NULL,NULL,NULL,
+    NULL,NULL,NULL,NULL,
+    NULL,NULL,
+    NULL,
+    NULL,NULL,NULL,
+    NULL,NULL,
+    NULL,NULL,
+    [],(NULL,NULL),map(),
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL
+),
+(
+    2,2,2,2,
+    2,2,2,2,
+    nan,
+    nan,
+    999.9999,
+    '2025-09-23',
+    '2025-09-23 01:02:03',
+    '2025-09-23 01:02:03.321',
+    'xyz','zzzz','b','y',
+    [4,5],
+    (2,'u'),
+    map('k2',2),
+    200,
+    generateUUIDv4(),
+    '10.0.0.1',
+    '2001:db8::1',
+    '{"k2":"v2"}'
+),
+(
+    -5,-5,-5,-5,
+    255,65535,4294967295,18446744073709551615,
+    -1,0.0,
+    -123.0001,
+    '1970-01-01',
+    '1970-01-01 00:00:00',
+    '1970-01-01 00:00:00.000',
+    '', 'aaaa','a','x',
+    [],(0,''),
+    map('edge',-1),
+    NULL,
+    generateUUIDv4(),
+    '0.0.0.0',
+    '::',
+    '{}'
+);
+
+SELECT '1.1 Basic NULL-safe equality and distinctness';
+SELECT 'Compare same column to itself and cross type';
+SELECT
+    c_int8 <=> c_int8      AS int8_num_self,
+    c_int8 IS DISTINCT FROM c_int8 AS int8_num_self_distinct,
+    c_float32 <=> c_float32 AS float_self_cmp,
+    c_float32 IS DISTINCT FROM c_float32 AS float_self_distinct,
+    c_nullable <=> NULL AS nullable_vs_null,
+    c_nullable IS DISTINCT FROM NULL AS nullable_vs_null_distinct,
+    c_float32 <=> c_float64 AS float_cross,
+    c_float32 IS DISTINCT FROM c_float64 AS float_cross_distinct
+FROM 03611_nscmp_tbl
+ORDER BY c_int32;
+
+SELECT '1.2 NaN behavior';
+SELECT 'NaN <=> NaN = 0, NaN <=> number = 0';
+SELECT 'NaN is distinct from NaN = 1, NaN is distinct from number = 1';
+SELECT
+    c_float32,
+    c_float32 <=> nan AS nan_cmp,
+    c_float32 IS DISTINCT FROM nan AS nan_distinct,
+    c_float32 <=> 1.0 AS cmp_with_one,
+    c_float32 IS DISTINCT FROM 1.0 AS cmp_with_one_distinct
+FROM 03611_nscmp_tbl
+ORDER BY c_int32;
+
+SELECT '1.3 String, FixedString, Enum';
+SELECT
+    c_string <=> 'abc' AS string_vs_literal,
```

Comment:
I would love to see comparison between all of these (strings, enums, date times, complex types, etc.) type and NULL values of the same type.